modules/tide_jira/src/TideJiraAPI.php

```diff
@@ -0,0 +1,290 @@
+    $this->tideSiteHelper = $site_helper;
+    $this->queueBackend = $queue_backend->get(self::QUEUE_NAME);
+    $this->entityTypeManager = $entity_manager;
+    $this->dateFormatter = $date_formatter;
+    $this->logger = $logger->get('tide_jira');
+  }
+
+  /**
+   * Extracts relevant metadata from a Node and queues a Jira ticket.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   A node entity.
+   */
+  public function generateJiraRequest(NodeInterface $node) {
+    $author = $this->getAuthorInfo($node);
+    if (!empty($author)) {
+      $revision = $this->getRevisionInfo($node);
+      $summary = $this->getSummary($revision);
+      $description = $this->templateDescription($author['name'], $author['email'], $author['department'], $revision['title'], $revision['id'], $revision['moderation_state'], $revision['bundle'], $revision['is_new'], $revision['updated_date'], $revision['notes'], $revision['preview_links']);
+      $request = new TideJiraTicketModel($author['name'], $author['email'], $author['department'], $revision['title'], $summary, $revision['id'], $revision['moderation_state'], $revision['bundle'], $revision['is_new'], $revision['updated_date'], $author['account_id'], $description, $author['project'], $revision['preview_links']);
+      $this->queueBackend->createItem($request);
+      $this->logger->debug('Queued support request for user ' . $author['email'] . ' for page ' . $revision['title']);
+    }
+    else {
+      $this->logger->notice('User ' . $node->getRevisionUser()->getEmail() . ' has no department/project set.');
+    }
+  }
+
+  /**
+   * Generates frontend links for a node.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node.
+   * @param bool $stringify
+   *   Whether to return an array or concatenated string.
+   *
+   * @return array|string
+   *   Preview links.
+   */
+  private function getPreviewLinks(NodeInterface $node, $stringify = FALSE) {
+    $results = [];
+    $sites = $this->tideSiteHelper->getEntitySites($node);
+    $sites = $sites['ids'];
+
+    foreach ($sites as $site_id) {
+      $term = $this->entityTypeManager->getStorage('taxonomy_term')->load($site_id);
+      $result = $this->tideSitePreviewHelper->buildFrontendPreviewLink($node, $term);
+      array_push($results, $result['url']->getUri());
+    }
+
+    if ($stringify) {
+      $temp = '';
+      foreach ($results as $key => $result) {
+        if (!($key === array_key_last($results))) {
+          $temp .= $result . ', ';
+        }
+        else {
+          $temp .= $result;
+        }
+      }
+      $results = $temp;
+    }
+    return $results;
+  }
+
+  /**
+   * Returns field_jira_project for a taxonomy.
+   *
+   * @param int $tid
+   *   Taxonomy term ID.
+   *
+   * @return string
+   *   Returns value of field_jira_project.
+   */
+  private function getProjectInfo($tid) {
+    $dept = $this->entityTypeManager->getStorage('taxonomy_term')->load($tid);
```

Comment:
I think you need to check the $tid existence here as well. What it passes the NULL value